app/services/products/fetch_product_calories.rb

```diff
@@ -0,0 +1,25 @@
+class Products::FetchProductCalories
+
+  def initialize(product)
+    @product = product
+  end
+
+  def call
+    fetch_product_calories
+  end
+
+  private
+
+  def fetch_product_calories
+    nutritionix_product = Nutritionix::ApiClient.new
+    product_info = nutritionix_product.get_product_data(@product.name)
+    if product_info.present?
```

Comment:
instead of wrapping everything in `if` statement you can write `return unless product_info.present?`. but this is something which probably rubocop will point later so you can keep it like it is for now :) it's called "guard clause"